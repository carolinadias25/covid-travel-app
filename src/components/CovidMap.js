/* eslint-disable */
import React, { memo, useEffect, useState } from "react";
import {
  ZoomableGroup,
  ComposableMap,
  Geographies,
  Geography,
} from "react-simple-maps";

import axios from "axios";
import "../styles/CovidMap.scss";
import { useHistory } from "react-router-dom";

const geoUrl =
  "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";

const CovidMap = ({ setTooltipContent }) => {
  const [riskFactor, setRiskFactor] = useState(undefined);
  const [countryName, setCountryName] = useState("");

  const history = useHistory();

  useEffect(() => {
    axios
      .request({
        method: "GET",
        url: "https://www.travel-advisory.info/api",
      })
      .then((response) => {
        setRiskFactor(response.data);
      })
      .catch((error) => {
        // eslint-disable-next-line no-console
        console.log(error);
      });
  }, []);

  const getHoverColor = (geo) => {
    const { ISO_A2 } = geo.properties;
    if (!riskFactor) {
      return "#D6D6DA";
    }
    const riskFactorData = riskFactor.data[ISO_A2];
    if (riskFactorData) {
      const { score } = riskFactorData.advisory;
      if (score >= 4.5) {
        return "#F53";
      }
      if (score >= 3.5) {
        return "#ffa442";
      }
      if (score < 3.5) {
        return "#3f5";
      }
    }
    return "#D6D6DA";
  };

  return (
    <div className="covidMap-page">
      <p className="covidMap-title">COVID-19 ZOOMABLE MAP</p>
      <p className="covidMap-info">
        Click and drag to pan map ⬩ Hover over country to see risk factor ⬩
        Click on country for further COVID-19 info
      </p>
      <div className="covidMap-BigContainer">
        <div className="covidMapContainer">
          <ComposableMap
            className="Map"
            data-tip=""
            projectionConfig={{ scale: 210 }}
          >
            <ZoomableGroup>
              <Geographies geography={geoUrl}>
                {({ geographies }) =>
                  geographies.map((geo) => (
                    <Geography
                      className="info-message"
                      key={geo.rsmKey}
                      geography={geo}
                      onClick={() => {
                        if (countryName === "Dem. Rep. Congo") {
                          history.push(`/country-info/DRC`);
                        } else if (countryName === "Central African Rep.") {
                          history.push(`/country-info/CAR`);
                        } else if (countryName) {
                          history.push(`/country-info/${countryName}`);
                        }
                      }}
                      onMouseEnter={() => {
                        const { NAME, ISO_A2 } = geo.properties;
                        const message = riskFactor.data[ISO_A2]
                          ? `Risk Factor: ${riskFactor.data[ISO_A2].advisory.score}/5`
                          : "Data unavailable";
                        setTooltipContent(`${NAME} - ${message}`);
                        setCountryName(NAME);
                      }}
                      onMouseLeave={() => {
                        setTooltipContent("");
                        setCountryName("");
                      }}
                      style={{
                        default: {
                          fill: "#D6D6DA",
                          outline: "none",
                        },
                        hover: {
                          fill: getHoverColor(geo),
                          outline: "none",
                        },
                        pressed: {
                          fill: "#E42",
                          outline: "none",
                        },
                      }}
                    />
                  ))
                }
              </Geographies>
            </ZoomableGroup>
          </ComposableMap>
        </div>
      </div>
    </div>
  );
};

export default memo(CovidMap);
