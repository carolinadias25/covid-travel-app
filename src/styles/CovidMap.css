html {
  font-family: system-ui, sans-serif;
  font-size: 1.25rem;
}

body {
  padding: 0;
  margin: 3;
  width: 100%;
  object-fit: cover;
}

svg {
  display: inline-block;
  vertical-align: middle;
}

.covidMapContainer {
  width: 100vw;
  height: 100vh;
  /* calc(100vh - 100px); */
}
