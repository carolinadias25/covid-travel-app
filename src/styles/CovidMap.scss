html {
  font-family: system-ui, sans-serif;
  font-size: 25px;
}
.covidMap-page {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: auto;
  width: 1500px;
  height: 800px;
  vertical-align: middle;
}
.covidMap-BigContainer {
  align-self: center;
  display: flex;
  justify-content: center;
  margin: 0;
  width: 1305px;
  height: 700px;
  background: #e5e5e5;
  -webkit-box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.25);
  box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.25);
  border-radius: 15px;
}
.covidMapContainer {
  align-self: center;
  display: flex;
  justify-content: center;
  width: 1100px;
  height: 500x;
  left: 133px;
  top: 226px;

  background: #ffffff;
  box-shadow: 3px 4px 4px rgba(0, 0, 0, 0.15);
  border-radius: 15px;
}

svg {
  align-self: center;
  display: flex;
  justify-content: center;
  width: 1200px;
  height: 533px;
  margin-top: auto;
}

.info-message {
  font-family: Nunito Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
}
