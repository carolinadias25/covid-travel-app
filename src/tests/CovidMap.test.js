/* eslint-disable */
import React from "react";
import { render } from "@testing-library/react";
import CovidMap from "../components/CovidMap";

describe("CovidMap", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<CovidMap />);

    expect(asFragment()).toMatchSnapshot();
  });
});
