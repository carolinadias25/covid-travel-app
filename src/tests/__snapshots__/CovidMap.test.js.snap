// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CovidMap renders correctly 1`] = `
<DocumentFragment>
  <div
    class="covidMap-page"
  >
    <p
      class="covidMap-title"
    >
      COVID-19 ZOOMABLE MAP
    </p>
    <p
      class="covidMap-info"
    >
      Click and drag to pan map ⬩ Hover over country to see risk factor ⬩ Click on country for further COVID-19 info
    </p>
    <div
      class="covidMap-BigContainer"
    >
      <div
        class="covidMapContainer"
      >
        <svg
          class="rsm-svg Map"
          data-tip=""
          viewBox="0 0 800 600"
        >
          <g>
            <rect
              fill="transparent"
              height="600"
              width="800"
            />
            <g
              class="rsm-zoomable-group undefined"
              transform="translate(0 0) scale(1)"
            >
              <g
                class="rsm-geographies "
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
